src/Language/PureScript/Scheme/CodeGen/Transpiler.hs

```diff
@@ -235,32 +235,34 @@ moduleToLibrary (Module _sourceSpan _comments moduleName _path
     caseToScheme values caseAlternatives =
       cond clauses
       where
-        clauses = map condClause alternatives
+        clauses = concatMap condClause alternatives
         alternatives = toAlternatives values caseAlternatives
 
-        -- Emit a pair (condition, result) for each case alternative to be used
-        -- by cond.
-        condClause :: Alternative -> (SExpr, SExpr)
+        -- Emit a list of pairs (condition, result) for each case alternative to be used
+        -- by cond. Every case alternative can produce multiple of them because there
+        -- might be multiple guards for each of them.
+        condClause :: Alternative -> [(SExpr, SExpr)]
 
-        -- If the result is a guarded expression then the result of the clause
-        -- is a nested cond expression.
+        -- If the result is a guarded expression then we concatenate the guard
```

Comment:
Concatenation happens in caseToScheme, here we just generate a list of clauses.

Reply:
No, the concatenation happens here because produce a `SExpr` for the guard, and we pass it to `test`